src/multivariate/multinomial.jl

```diff
@@ -118,16 +122,18 @@ end
 function _logpdf{T<:Real}(d::Multinomial, x::AbstractVector{T})
     p = probs(d)
     n = ntrials(d)
-    s = lgamma(n + 1.0)
-    t = 0
+    S = eltype(p)
+    R = promote_type(T, S)
+    s = R(lgamma(n + 1))
+    t = zero(T)
     for i = 1:length(p)
         @inbounds xi = x[i]
-        @inbounds pi = p[i]
+        @inbounds p_i = p[i]
         t += xi
-        s -= lgamma(xi + 1.0)
-        @inbounds s += xi * log(pi)
+        s -= R(lgamma(xi + 1))
```

Comment:
`lgamma(Float32(1) + 1)` is `Float32`
`lgamma(Float16(1) + 1)` is `Float32`

If `T` is `Int` and `S` is `Float32`, then `R` is `Float32` but `lgamma(xi + 1)` is `Float64`.
That's a bit of a weird edge case, but it could come up.